be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/LaoValidator.scala

```diff
@@ -27,6 +32,8 @@ case object LaoValidator extends MessageDataContentValidator {
           Right(validationError("unexpected id"))
         } else if (data.organizer != message.sender) {
           Right(validationError("unexpected organizer public key"))
+        } else if (channel != Channel.ROOT_CHANNEL) {
+          Right(validationError(s"trying to send a CreateLao message on a wrong type of channel $channel"))
```

Comment:
Error message should be more specific what about: "trying to send a CreateLao message on $channel other than ${Channel.ROOT_Channel}"

Reply:
I'll add this but I'm wondering whether this would leak implementation details in theory (even though everyone is allowed to create a LAO so should not be a problem here)